posts/2023-11-28-23.0.0.12-beta.adoc

```diff
@@ -0,0 +1,253 @@
+# Do NOT change the categories section
+categories: blog
+author_picture: https://avatars3.githubusercontent.com/lauracowen
+author_github: https://github.com/lauracowen
+seo-title: Preview of Jakarta Data Milestone 1 in Open Liberty 23.0.0.12-beta - OpenLiberty.io
+seo-description: Annotate lifecycle methods in a standardized way with Jakarta Data. Try Milestone 1 now and give feedback on the spec so far.
+blog_description: Annotate lifecycle methods in a standardized way with Jakarta Data. Try Milestone 1 now and give feedback on the spec so far.
+open-graph-image: https://openliberty.io/img/twitter_card.jpg
+open-graph-image-alt: Open Liberty Logo
+---
+= Preview of Jakarta Data Milestone 1 in Open Liberty 23.0.0.12-beta
+Laura Cowen <https://github.com/lauracowen>
+:imagesdir: /
+//Blank line here is necessary before starting the body of the post.
+
+Open Liberty 23.0.0.12-beta contains a preview of the new Jakarta Data specification for Jakarta EE, as it currently stands at Milestone 1. Milestone 1 provides the capability to annotate lifecycle methods such as `Insert`, `Delete`, and more. You can try it out and give feedback on the specification so far.
+
+In this beta we have also added the ability to configure the duration of the _quiesce_ stage when the Liberty runtime shutsdown, for situations in which services need longer than the default 30 seconds to complete the processing of requests.
+
+The link:/[Open Liberty] 23.0.0.12-beta includes the following beta features (along with link:/docs/latest/reference/feature/feature-overview.html[all GA features]):
+
+* <<jdata, Preview of Jakarta Data (Milestone 1)>>
+* <<timeout, Configurable Quiesce Timeout>>
+
+See also link:/blog/?search=beta&key=tag[previous Open Liberty beta blog posts].
+
+// // // // DO NOT MODIFY THIS COMMENT BLOCK <GHA-BLOG-TOPIC> // // // // 
+// Blog issue: https://github.com/OpenLiberty/open-liberty/issues/26759
+// Contact/Reviewer: njr-11
+// // // // // // // // 
+[#jdata]
+== Preview of Jakarta Data (Milestone 1)
+
+Jakarta Data is a new link:https://jakarta.ee/specifications/data/1.0/[Jakarta EE specification] being developed in the open that aims to standardize the popular data link:https://deviq.com/design-patterns/repository-pattern[repository pattern] across a variety of providers. Open Liberty includes the Jakarta Data 1.0 Milestone 1 release, which adds the ability to annotatively compose custom lifecycle methods, covering `Insert`, `Update`, `Save`, and `Delete` operations. The Open Liberty beta includes a test implementation of Jakarta Data that we are using to experiment with proposed specification features so that developers can try out these features and provide feedback to influence the Jakarta Data 1.0 specification as it continues to be developed after Milestone 1. The test implementation currently works with relational databases and operates by redirecting repository operations to the built-in Jakarta Persistence provider.
```

Comment:
I don't think we should link to another vendor's documentation here. After the initial paragraph it gets into specifics of that vendor's own solution.
```suggestion
Jakarta Data is a new link:https://jakarta.ee/specifications/data/1.0/[Jakarta EE specification] being developed in the open that aims to standardize the popular data repository pattern across a variety of providers. Open Liberty includes the Jakarta Data 1.0 Milestone 1 release, which adds the ability to annotatively compose custom lifecycle methods, covering `Insert`, `Update`, `Save`, and `Delete` operations. The Open Liberty beta includes a test implementation of Jakarta Data that we are using to experiment with proposed specification features so that developers can try out these features and provide feedback to influence the Jakarta Data 1.0 specification as it continues to be developed after Milestone 1. The test implementation currently works with relational databases and operates by redirecting repository operations to the built-in Jakarta Persistence provider.
```